From jrm8005@gmail.com  Tue Oct 21 15:59:50 2008
From: "J.R. Mauro" <jrm8005@gmail.com>
Date: Mon, 20 Oct 2008 20:36:34 -0400
Subject: Staging: ppc-acpi: Handle errors from input_register_device()
To: Greg KH <gregkh@suse.de>
Message-ID: <20081021003634.GA1267@localhost>
Content-Disposition: inline

From: J.R. Mauro <jrm8005@gmail.com>

Handle errors from input_register_device()

Signed-off by: J.R. Mauro <jrm8005@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/pcc-acpi/pcc-acpi.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/drivers/staging/pcc-acpi/pcc-acpi.c
+++ b/drivers/staging/pcc-acpi/pcc-acpi.c
@@ -842,6 +842,7 @@ static int acpi_pcc_init_input(struct ac
 {
 	struct input_dev *hotk_input_dev;
 	struct pcc_keyinput *pcc_keyinput;
+	int ret;
 
 	ACPI_FUNCTION_TRACE("acpi_pcc_init_input");
 
@@ -885,8 +886,13 @@ static int acpi_pcc_init_input(struct ac
 
 	hotkey->input_dev = hotk_input_dev;
 
+	ret = input_register_device(hotk_input_dev);
 
-	input_register_device(hotk_input_dev);
+	if (ret) {
+		input_free_device(hotk_input_dev);
+		kfree(pcc_keyinput);
+		return ret;
+	}
 
 	return_VALUE(0);
 }
